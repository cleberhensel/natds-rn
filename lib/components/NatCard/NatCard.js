import React from 'react';
import { ScrollView } from 'react-native';
import PropTypes from 'prop-types';
import { NatText } from '../NatText';
import { NatImage } from '../NatImage';
import { CardWrapper, CardFooter } from './NatCard.styles';

const defaultMargin = (propMargin) => {
  switch (propMargin) {
    case 'none': return '0';
    case 'dense': return '8';
    case 'normal': return '16';
    default: return '0';
  }
};

export const NatCard = ({
  onPress, type, title, subtitle, imageURI, resume, margin, accessibilityLabel, accessibilityHint, imageDescription,
}) => (
  <ScrollView>

    <CardWrapper
      onPress={onPress}
      defaultMargin={defaultMargin(margin)}
      accessible
      accessibilityLabel={accessibilityLabel}
      accessibilityHint={accessibilityHint}
    >
      <NatImage
        source={imageURI}
        type="card"
      />
      <CardFooter>
        <NatText
          type="h6"
          value={title}
        />
        <NatText
          type="caption"
          value={subtitle}
        />
        <NatText
          type="body2"
          value={resume}
          margin="dense"
        />
      </CardFooter>
    </CardWrapper>
  </ScrollView>

);

NatCard.propTypes = {
  type: PropTypes.string,
  onPress: PropTypes.func.isRequired,
  title: PropTypes.string.isRequired,
  subTitle: PropTypes.string,
  imageURI: PropTypes.string.isRequired,
  resume: PropTypes.string.isRequired,
  margin: PropTypes.string,
  accessibilityLabel: PropTypes.string,
  accessibilityHint: PropTypes.string,
  imageDescription: PropTypes.string,
};

NatCard.defaultProps = {
  type: 'default',
  title: 'Title',
  subTitle: 'SubTitle',
};
