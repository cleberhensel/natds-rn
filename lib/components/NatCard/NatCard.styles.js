import styled from 'styled-components/native';
import { colors } from '../../assets/theme/colors/colors';

const inputStyles = {
  default: {
    backgroundColor: colors.primary.white,
  },
};

const getStylesFrom = styles => type => styles[type] || styles.default;

const getInputStyles = getStylesFrom(inputStyles);

export const CardWrapper = styled.TouchableOpacity`
  background-color: ${({ type }) => getInputStyles(type).backgroundColor};
  border-radius: 4px;
  margin-bottom: ${({ defaultMargin }) => defaultMargin || '0px'};
`;

export const CardFooter = styled.View`
  padding: 10px;
`;
