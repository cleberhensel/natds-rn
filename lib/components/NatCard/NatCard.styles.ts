import styled from "styled-components/native";
import { ThemeNaturaLight } from "../../themes/ThemeNaturaLight";
const inputStyles = {
  default: {
    backgroundColor: ThemeNaturaLight.palette.background.paper
  }
};
const getStylesFrom = styles => type => styles[type] || styles.default;
const getInputStyles = getStylesFrom(inputStyles);
export const CardWrapper = styled.TouchableOpacity`
  background-color: ${({ type }) => getInputStyles(type).backgroundColor};
  border-radius: 4px;
  margin-bottom: ${({ defaultMargin }) => defaultMargin || "0px"};
`;
export const CardFooter = styled.View`
  padding: 10px;
`;
