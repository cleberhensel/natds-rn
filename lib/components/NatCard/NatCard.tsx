import React from 'react';
import {
  View,
} from 'react-native';
import PropTypes from 'prop-types';
import { NatText } from '../NatText';
import { NatImage } from '../NatImage';
import { CardWrapper, CardFooter } from './NatCard.styles';
import {
  ThemeNaturaLight,
} from '../../themes/ThemeNaturaLight';

const defaultMargin = (propMargin) => {
  switch (propMargin) {
    case 'micro': return '4';
    case 'tiny': return '8';
    case 'small': return '16';
    case 'standard': return '24';
    default: return '0';
  }
};

const styles = {
  borderRadius: 4,
  elevation: 1,
  shadowColor: '#000',
  shadowOffset: {
    height: 1,
    width: 0,
  },
  shadowOpacity: 0.24,
  shadowRadius: 4,
};

export const NatCard = ({
  onPress,
  title,
  subtitle,
  imageURI,
  resume,
  margin,
  accessibilityLabel,
  accessibilityHint,
  theme,
}) => (
  <View style={styles}>
    <CardWrapper
      onPress={onPress}
      defaultMargin={defaultMargin(margin)}
      accessible
      accessibilityLabel={accessibilityLabel}
      accessibilityHint={accessibilityHint}
    >
      <NatImage
        source={imageURI}
        type="card"
      />
      <CardFooter>
        <NatText
          type="h6"
          value={title}
          color={theme.palette.text.primary}
          margin={theme.spacing.spacingMicro}
        />
        <NatText
          type="caption"
          value={subtitle}
          color={theme.palette.text.secondary}
          margin={theme.spacing.spacingSmall}
        />
        <NatText
          type="body2"
          value={resume}
          color={theme.palette.text.primary}
          margin={theme.spacing.spacingStandard}
        />
      </CardFooter>
    </CardWrapper>
  </View>

);

NatCard.propTypes = {
  accessibilityHint: PropTypes.string,
  accessibilityLabel: PropTypes.string,
  imageURI: PropTypes.string.isRequired,
  margin: PropTypes.string,
  onPress: PropTypes.func.isRequired,
  resume: PropTypes.string.isRequired,
  subtitle: PropTypes.string,
  theme: PropTypes.shape({}),
  title: PropTypes.string.isRequired,
};

NatCard.defaultProps = {
  subTitle: 'SubTitle',
  theme: ThemeNaturaLight,
  title: 'Title',
  type: 'default',
};
