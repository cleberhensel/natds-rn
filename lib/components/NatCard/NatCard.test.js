import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components/native';

import { NatCard } from './NatCard';

describe('TextInput component default', () => {
  it('Should render textInput component ', () => {
    const button = renderer.create(
      <NatCard
        title="teste"
        onPress={jest.fn()}
        imageURI="https://upload.wikimedia.org/wikipedia/pt/thumb/c/cb/Natura_Logo.png/200px-Natura_Logo.png"
        resume="teste"
      />,
    ).toJSON();
    expect(button).toMatchSnapshot();
  });
});
