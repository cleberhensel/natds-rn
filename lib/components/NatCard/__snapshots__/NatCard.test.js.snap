// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput component default Should render textInput component  1`] = `
<RCTScrollView>
  <View>
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#FFFFFF",
          "borderRadius": 4,
          "marginTop": 0,
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "width": "100%",
            },
          ]
        }
        type="card"
      >
        <Image
          source={
            Object {
              "uri": "https://upload.wikimedia.org/wikipedia/pt/thumb/c/cb/Natura_Logo.png/200px-Natura_Logo.png",
            }
          }
          style={
            Array [
              Object {
                "borderRadius": 0,
                "height": 184,
                "width": "100%",
              },
            ]
          }
          type="card"
        />
      </View>
      <View
        style={
          Array [
            Object {
              "paddingBottom": 10,
              "paddingLeft": 10,
              "paddingRight": 10,
              "paddingTop": 10,
            },
          ]
        }
      >
        <View
          defaultMargin="0px"
          style={
            Array [
              Object {
                "flexBasis": 0,
                "flexGrow": 1,
                "flexShrink": 1,
                "marginTop": 0,
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#333333",
                  "fontSize": 20,
                  "fontWeight": "400",
                  "lineHeight": 24,
                },
              ]
            }
            type="h6"
          >
            teste
          </Text>
        </View>
        <View
          defaultMargin="0px"
          style={
            Array [
              Object {
                "flexBasis": 0,
                "flexGrow": 1,
                "flexShrink": 1,
                "marginTop": 0,
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#777777",
                  "fontSize": 12,
                  "fontWeight": "400",
                  "lineHeight": 14,
                },
              ]
            }
            type="caption"
          />
        </View>
        <View
          defaultMargin="8px"
          style={
            Array [
              Object {
                "flexBasis": 0,
                "flexGrow": 1,
                "flexShrink": 1,
                "marginTop": 8,
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#777777",
                  "fontSize": 14,
                  "fontWeight": "400",
                  "lineHeight": 21,
                },
              ]
            }
            type="body2"
          >
            teste
          </Text>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
